### DISCUSSION TITLE
Fatal log messages have debug log level

### DISCUSSION THREAD

**Discussion by ghost** [renovatebot/renovate#21694] [Suggest an Idea]
**What Renovate type, platform and version are you using?**

Selfhosted

**Describe the bug**

When a push fails, the failure is logged with the `debug` log level. Not pushing means the bot isn't working, and I think that warrants a `fatal`, or `error` log, not `debug`.

Additionally, after such failure, exit code is still 0. This makes it hard to detect that the bot is no longer working (e.g. if it no longer has permissions). Failure should result in a non-zero exit code.

**Relevant debug logs**

```
DEBUG: Error commiting files (repository=*/*, branch=renovate/system.io.abstractions-12.x)
       "err": {
         "task": {
           "commands": [
             "push",
             "origin",
             "renovate/system.io.abstractions-12.x:renovate/system.io.abstractions-12.x",
             "--force",
             "-u",
             "--no-verify",
             "--verbose",
             "--porcelain"
           ],
           "format": "utf-8"
         },
         "message": "Pushing to https://*.visualstudio.com/*/*/_git/*\nPOST git-receive-pack (883 bytes)\nerror: failed to push some refs to 'https://*.visualstudio.com/*/*/_git/*'\n",

```

**To Reproduce**

**Additional context**

**Comment by rarkins** [comment]
I think a warn here is appropriate, not fatal or error
